### DISCUSSION TITLE
Add dual Mic input functionality when using Media Mod.

### DISCUSSION THREAD

**Discussion by gregb886** [gopro/labs#329] [Feature Requests]
Request for a new firmware feature:
I use my GP10 Black & GP8 Black in a rally car and I want to record both the helmet intercom audio out Mono RCA cable to L&R channels of a 3.5mm stereo plug and the car engine noise.  The intercom audio is the co-driver calling rally notes to the driver with occasional driver responses.  
If I connect the intercom cable to the Media Mod, I get excellent intercom audio but almost no car sound because the mics are inside our helmets.  
The car is noisy and if I just use the Media Mod mic with no cable from the intercom, I can't hear the co-driver on the video.  I need both audio inputs combined.
I am trying various work arounds but it would be nice to have a Mod setting that allowed both the Media Mod 3.5mm audio socket with the Mic & Mic + options and the Media Mod Mic with the Mic & Mic + options to have both audio inputs combined with a choice of input level settings.
Thanks to those who have been helping me with the work arounds.

**Comment by dnewman-gpsw** [comment]
Same solution from the GoPro forum is what works now: "Audio Line-in is stereo. So send your helmet microphone to left channel and add an environmental microphone on the right channel. Then in post you can mix to the desired levels."  This is info for others with similar questions.

The suggested solutions are not likely to happen in Labs, as it would involve too deep changes to the audio processing engine. Labs enables or modifies existing processing paths, this requires new engineering. This is a feature request for a future camera.

**Reply by gregb886** [reply]
Thank you for responding, David
I will test both these methods:
1. helmet microphone to left channel and add an environmental microphone on the right channel (will require post to balance the audio)
2. helmet microphone & environment microphone with normal or +20db switch to a 2 stereo sockets to 1 stereo plug adapter (hopefully that switch will give me enough control to not require post)

I have the right cable for method #1.  I have tested it and it works perfectly. 

I would be a very happy customer if a future camera included this new feature.  Thanks

**Reply by cprima** [reply]
The hobbyist audio-engineer-apprentice in me would love to get access to a file of the 1st variant!
Because I am sure a "quick&dirty" workflow for postproduction that incorporates "ducking" will greatly enhance the result. (While the helmet microphones are spoken into the level of the environment track should be reduced.)

And of course there is the other solution: Use 2 cameras (and combine those in editing). ;)

**Reply by gregb886** [reply]
Thanks for your reply b
We are competing at an event this weekend and will test option 1. 
I will put a file on Google drive & share it late next week when I am back home. 
My primary purpose is to avoid having to post after an event. We do many stages and post is very time consuming. 
I have been using two cameras and RaceRender 3 to synchronise the two videos and adjust the audio. 
I now have my 10 configured with Labs firmware to give satisfactory video from between the seats to show the in car action and the road ahead well. 
If I can get the audio working then no more post!

**Reply by gregb886** [reply]
I did not understand what Ducking is but now that I have read your reply again and researched it I am interested.  How would I achieve that given that helmet audio is on L channel and environment is on R channel?

**Reply by cprima** [reply]
> I did not understand what Ducking is but now that I have read your reply again and researched it I am interested. How would I achieve that given that helmet audio is on L channel and environment is on R channel?

When I wrote this yesterday I made a quick test with the (free as in beer) Windows/Mac/Linux software Audacity https://www.audacityteam.org/ , but with "real footage" from you I would gladly document the process -- the steps should be repeatably in various software tools.

Will be on a bicycle tour this and next weekend, but certainly can give it a try over the course of next weeks' weekdays! I understand that you might have footage available from about May 11th on.
I put my email into my GitHub profile: https://github.com/cprima , for either file sharing or to ping me should I miss a notificytion from this thread).

I would think this is the way to go in Audacity:
- split stereo into 2 mono channels https://manual.audacityteam.org/man/splitting_and_joining_stereo_tracks.html
- place the environment track on top of the intercom track (drag&drop)
- apply from effect menu "auto-duck" https://manual.audacityteam.org/man/auto_duck.html
- export the audio as e.g. wav file

Here is a proof of concept, without any loudness normalization (for YouTube) or similar,
excuse the somewhat ugly tone-generator sound: https://youtu.be/ULnm5hO-7C0
All steps are conveniently contained in a **macro** https://gist.github.com/cprima/07c0b530c04aab6432ace33526b223b0  :) (software automation is my profession ;) )

On top of that I'd love to do some loudness normalization and look at the waveforms of the audio for further ideas. 
(I am not an expert in audio engineering but certainly did educate myself over the past years.)

_By the way: The loudness issues with online videos are nicely explained here https://youlean.co/how-to-edit-a-video-to-achieve-good-audio-loudness-on-youtube/ (but can also be achieved with Audacity)._


**I fully understand the time constraints for editing, they are the similar on a bicycle tour!**
Maybe this process is only good for a "season best-of". Still would love to tinker with your footage: Fun and educative for me, and maybe an inspiration/aspiration/"ear"opener for you!

@gregb886 I would have a request: Please
- make an additional ~10 second recording
- with both microphones inserted
- camera mounted in the car
- no one talking into the helmet microphones
- and the environment as silent as possible, e.g. engine off (maybe even at home in the garage?)
Such a recording can be used to determine the "noise floor" of the recording equipment, or to masquerade parts of the video without any audio from the footage.

**Reply by gregb886** [reply]
Thanks for the time you have put into this.
I am not having any success getting the helmet into L channel and ext mic into R channel.  I am using a dedicated cable that has a black 3.5 mono socket and a red 3.5 mono socket that combine to a 3.5 stereo plug that goes to my GP10 Media Mod.
When I load the mp4 file into Audacity I can see the L&R channels are identical.
This is the cable I am using:
Hosa YMM-261 3.5mm TRS(M) to Dual 3.5mm TS(F) Stereo Breakout Adaptor Cable
https://www.mannys.com.au/hosa-ymm-261-3-5mm-trs-m-to-dual-3-5mm-ts-f-stereo-breakout-adaptor-cable?utm_medium=google-free-shopping-listings&utm_source=RQmedia

Is there an Audio setting I need to change on the GoPro to get separate channels?  Or is my cable setup the problem?

**Reply by moll** [reply]
@dnewman-gpsw:
> The suggested solutions are not likely to happen in Labs, as it would involve too deep changes to the audio processing engine. Labs enables or modifies existing processing paths, this requires new engineering. This is a feature request for a future camera.

Hey, the original request was for in-device mixing, but somewhat relatedly, do you mind clarifying whether the raw audio feature (wav file) is supposed to save both the media mod mic _and_ the external mic into separate channels at the same time? I'm currently on the stock firmware on Hero 11 where it doesn't seem to, but trying figure out if that's possible at all and/or with Labs. Thanks!

**Reply by dnewman-gpsw** [reply]
Nothing added in Labs for this.

**Reply by moll** [reply]
Thanks, @dnewman-gpsw! You wouldn't happen to know if the functionality (saving the media mod mics in parallel with the external mic, all in separate channels) exists in the stock then at all? Thanks!

**Reply by dnewman-gpsw** [reply]
No idea.

**Comment by gregb886** [comment]
I have made the recording of the car background noise and a recording of the helmet and external mic audio and I have had some success.
Both recordings are here: https://drive.google.com/drive/folders/10kVDHVWTI6jk_ZstZTLaOBVdOqC3WpqW?usp=sharing

Here are my notes:
Rally car intercom to GoPro 10 Black
GP10 is set with Raw Audio off so I get MP4 files.
Preferred settings after testing many variations of mics, intercom & connections:
Helmet audio: Using car powered Stilo WRC DES intercom powered from the car battery with intercom audio out set to High.  Connected using a RCA to 3.5mm Mono cable with tested ground to ground and tip to tip connection. Plugged into the Red/Right mono connector of the HOSA dual mono to 3.5mm stereo cable to the GP10 Media Mod audio socket.
External Mic audio: Mic has a 3.5mm stereo plug and that is plugged into the Black/Left mono connector of the HOSA dual mono to 3.5mm stereo cable to the GP10 Media Mod audio socket.
Result = Preferred settings.mp4
In car with helmet & ext mic operating and car not running – 0- quiet recording with engine off.mp4

**Reply by cprima** [reply]
> I have made the recording of the car background noise and a recording of the helmet and external mic audio and I have had some success. Both recordings are here: https://drive.google.com/drive/folders/10kVDHVWTI6jk_ZstZTLaOBVdOqC3WpqW?usp=sharing
> 

Nice, on first sight in Audacity both channels are clearly different.
I was able to download both files.

I was on a bicycle tour to my 2nd place of residence over the weekend, and should be able from Thursday/Friday on to look into the easiest way to postprocess such files. (that's the issue with 2 apartments, some items are always missing… Out of curiosity I also ordered a cable by manufacturer RØDE with product name SC11 https://rode.com/en/accessories/adaptors-cables/sc11 which should also do what our cable does: "a TRS splitter cable for connecting two mono outputs to a single stereo input")

**Comment by gregb886** [comment]
Thanks Christian,
It is interesting that the Rode cable uses 2x TRS female sockets for mono.  My Hosa cable has 2x TS Mono female sockets to 1x TRS male plug.

The helmet to helmet intercom has a high & low output switch and the external mic has a Normal or +20db switch so I hope I can achieve a good balance without post but I do appreciate your advice.

Cheers

Greg Burrowes
0404 182 428

**Comment by gregb886** [comment]
**GoPro settings in rally car** 
GP10 on windscreen 
•	4K 30fps Linear
•	-1.0 exposure 
•	In Media Mod set to Standard Mic

Intercom:
•	Using Stilo WRC DES intercom with camera output on LOW
•	Intercom from power bank with 12v step up adapter to powered Stilo amp with intercom output switch on LOW to L channel of GP

Car sound external mic:
•	Car Mic with AAA battery to R channel 
•	Car mic on +20db

2x mono to 1x TRS stereo cable to GP Media Mod
•	BluTak Audio jack into GP media mod because it comes loose and creates static noise in the intercom helmet speakers.


GP8 between seats:
•	1080 30fps Narrow view 
•	-1.0 exposure with 
•	audio in from the Stilo WRC DES intercom on low output. 
•	This gets the whole dash including the RallySafe 


This setup gives good co-driver audio & car sound.
